StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFlowController.swift

```diff
@@ -295,6 +331,12 @@ extension PaymentSheet {
                     assertionFailure("The PaymentSheet.FlowController instance was destroyed during a call to `update(intentConfiguration:completion:)`")
                     return
                 }
+
+                // If we have an update in progress and this update call's ID does not match the latest update ID call, ignore this update
+                if case let .inProgress(latestUpdateID) = self.latestUpdateStatus, updateID != latestUpdateID {
```

Comment:
I don't believe we made any contract with our discussions internally that if an update fails we block or retry those previously failed updates. My understanding was that it was up to the merchant to retry if any update has failed.

Reply:
I meant to write more there, sorry - what I mean is, what happens if a newer update completed (failed or succeeded) when we get to this point?  I think we continue like normal, overwriting the newer update?

Like, we handle this case:
<img width="750" alt="image" src="https://user-images.githubusercontent.com/47796191/227385997-b460b1a0-5a15-404c-8b0f-879b9f7d0573.png">

But what about this case:
<img width="626" alt="image" src="https://user-images.githubusercontent.com/47796191/227386239-8586da1a-12bb-410b-bd44-c6a0c9cd313c.png">

Reply:
Hm yea, we'll just need to keep track of the last update ID to handle this edge case since in this case when update 1 finishes it'll see the last status as "completed" and finish.

The more I think on this though the more I wonder if the 1 thread approach is going to scale. We really should be queuing these one after another and not starting update 2 until we have a result for update 1, and invoking all the callbacks in-order, which is what I think the merchants will expect. The simple approach should work fine here but as we expand to working like "magic" I'm fud-ful the 1 thread approach will scale, especially when it comes to waiting for updates to finish.

Reply:
I don't think we need to wait until an update ends to start a new one, I would actually say we _shouldn't_ do that, it could cause the process to hang for a long time.
The update ID approach is okay, I've used it many times before, but we could use a queue and cancel an existing update before starting another one.

Reply:
Yeah I agree there must be some concurrency API(s) we can lean on to make this more robust/simpler for ourselves.  

A serial queue isn't quite right because we don't want previous updates to complete.  It'd make multiple updates take significantly longer. It'd also merchants to deal with the "what if update 1 failed after I started update 2" problem.  They'd need to add checks before re-trying a failed update to make sure they don't overwrite other updates.

I think what could make this simpler is if we encapsulated our update work into **cancelable** Operation (or perhaps Task?).  We'd keep track of the single current operation as `currentUpdateOperation`.  Whenever update is called, we call `currentUpdateOperation.cancel()` and replace it with a new update operation.

Reply:
A serial queue could work, even if we are not really using it as a real queue, we would just need to to cancel existing operations, but if we're gonna use it that way I agree that just keeping track of the current operation should be enough.